engine/collection/ingest/engine.go

```diff
@@ -65,13 +73,44 @@ func New(
 		},
 	)
 
+	// FIFO queue for transactions
+	queue, err := fifoqueue.NewFifoQueue(
+		fifoqueue.WithCapacity(int(config.MaxMessageQueueSize)),
+		fifoqueue.WithLengthObserver(func(len int) {
+			mempoolMetrics.MempoolEntries(metrics.ResourceTransactionIngestQueue, uint(len))
+		}),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not create transaction message queue: %w", err)
+	}
+	pendingTransactions := &engine.FifoMessageStore{FifoQueue: queue}
+
+	// define how inbound messages are mapped to message queues
+	handler := engine.NewMessageHandler(
+		logger,
+		engine.NewNotifier(),
+		engine.Pattern{
+			Match: func(msg *engine.Message) bool {
+				_, ok := msg.Payload.(*flow.TransactionBody)
+				if ok {
+					engMetrics.MessageReceived(metrics.EngineCollectionIngest, metrics.MessageTransaction)
+				}
+				return ok
+			},
+			Store: pendingTransactions,
+		},
+	)
+
 	e := &Engine{
-		unit:                 engine.NewUnit(),
+		lm:                   lifecycle.NewLifecycleManager(),
+		wg:                   new(sync.WaitGroup),
```

Comment:
rather than using a `LifecycleManager` and `WaitGroup`, maybe this could use a `ComponentManager`. Right now this mixes the interfaces and it may get confusing.

e.g. after engine is instantiated, add something like
```
	cm := component.NewComponentManagerBuilder().
		AddWorker(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
			ready()
			e.loop(ctx)
		}).
		Build()
	e.Component = cm
```

then you can ditch the explicit ready/done/start methods since the `Component` interface is provided by `cm`

Reply:
👍  Will do this, makes much more sense. Thanks!